libs/cardano-ledger-api/src/Cardano/Ledger/Api/Tx/Cert.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE PatternSynonyms #-}
+
+module Cardano.Ledger.Api.Tx.Cert (
+  EraTxCert,
+  TxCert,
+  getVKeyWitnessTxCert,
+  getScriptWitnessTxCert,
+  pattern RegPoolTxCert,
+  pattern RetirePoolTxCert,
+
+  -- * Shelley Era
+  ShelleyEraTxCert,
+  pattern MirTxCert,
+  pattern GenesisDelegTxCert,
+  pattern RegTxCert,
+  pattern UnRegTxCert,
+  pattern DelegStakeTxCert,
+
+  -- * Conway Era
+  ConwayEraTxCert,
+  Delegatee (..),
+  pattern RegDepositTxCert,
+  pattern UnRegDepositTxCert,
+  pattern DelegTxCert,
+  pattern RegDepositDelegTxCert,
+  pattern AuthCommitteeHotKeyTxCert,
+  pattern ResignCommitteeColdTxCert,
+  pattern RegDRepTxCert,
+  pattern UnRegDRepTxCert,
+)
+where
+
+import Cardano.Ledger.Conway.TxCert
```

Comment:
Just a haskell question: does this mean that the data constructors corresponding to these patterns (eg: ConwayRegCert, ConwayUnRegCert..etc) are also exported, or only the patterns?

Reply:
They are imported (due to open import), but they are not re-exported from the API. Not even the concrete type `ConwayTxCert` is exported from the API, because we don't want to promote multiple ways of doing the same thing. It is better to export the most general (thus more powerful) way to deal with things.

That question reminds me that I should probably do explicit imports as well, just to help people reading the code infer where things are coming from.

Reply:
Add explicit imports and improved haddock a bit.

Reply:
Awesome, thanks for the explanation!